nixos/modules/services/networking/xrdp.nix

```diff
@@ -59,10 +59,10 @@ in
       };
 
       port = mkOption {
-        type = types.port;
-        default = 3389;
+        type = types.enum [ types.port types.str ];
```

Comment:
Sounds like this will not work with `openFirewall`. It will probably be better to add a new option and make it mutually incompatible with port.